# HG changeset patch
# User pm
# Date 1526630767 -7200
#      Fri May 18 10:06:07 2018 +0200
# Node ID 5ccf9888ef044b35961dcff0a5791d11dc847203
# Parent  7f2121c801bba604b59b8ca59ad5a1085d27c486
Ticket mantis 0003269: Un lien mailto HTML dans balise <a> avec target fait planter le Courrielleur

diff --git a/comm-esr60/comm/mail/base/content/contentAreaClick.js b/comm-esr60/comm/mail/base/content/contentAreaClick.js
--- a/comm-esr60/comm/mail/base/content/contentAreaClick.js
+++ b/comm-esr60/comm/mail/base/content/contentAreaClick.js
@@ -151,8 +151,15 @@
   if (Cc["@mozilla.org/uriloader/external-protocol-service;1"]
         .getService(Ci.nsIExternalProtocolService)
         .isExposedProtocol(uri.scheme) &&
-      !uri.schemeIs("http") && !uri.schemeIs("https"))
+      !uri.schemeIs("http") && !uri.schemeIs("https")) {
+    // mantis 0003269: Un lien mailto HTML dans balise <a> avec target fait planter le Courrielleur
+    if (uri.schemeIs("mailto")) {
+      aEvent.preventDefault();
+      openURL(href);
+    }
+    //fin bug mantis 0003269
     return true;
+  }
 
   // Now we're here, we know this should be loaded in an external browser, so
   // prevent the default action so we don't try and load it here.
